### DISCUSSION TITLE
Vulnerabilities in log4j:log4j CVE-2022-23307 CVE-2022-23305 CVE-2022-23302 CVE-2021-4104 CVE-2019-17571

### DISCUSSION THREAD

**Discussion by liubagha** [jboss-logging/jboss-logging#111] [Show and tell]
It is a pleasure to work with your library, thank you very much. I would like to draw your attention to five direct vulnerabilities in the log4j:log4j dependency (https://mvnrepository.com/artifact/log4j/log4j/1.2.17).
Many projects move to the alternative recommended package org.apache.logging.log4j:log4j-core (https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-core).
Implementation of current version jboss-logging:jboss-logging becomes unsafe.

There is a warning about EOL of that https://mvnrepository.com/artifact/log4j/log4j/1.2.17 package:
[log4j](https://logging.apache.org/log4j/1.2/) (note this log manager is EOL’d)

What do you think about trying to use org.apache.logging.log4j:log4j-core package in your project?

**Comment by expertesantos** [comment]
Hi everyone, has there been any movement on this topic?

This library is used by the Quarkus framework, and AWS Amazon Inspector is flagging critical vulnerabilities in this dependency, which is blocking the use of Quarkus.

CVE-2019-17571, CVE-2022-23305, CVE-2023-26464, CVE-2022-23302, CVE-2022-23307, CVE-2021-4104, CVE-2020-9488.

I also opened a discussion about it there.

https://github.com/quarkusio/quarkus/discussions/43607

**Comment by dmlloyd** [comment]
This is a false report. `jboss-logging` can link with `log4j` but does not depend on it.

**Reply by expertesantos** [reply]
Thank you for the quick response, @dmlloyd 

Is there a plan or intention to remove log4j (v1) from the project's pom.xml?

**Reply by dmlloyd** [reply]
No, there is no plan to do so at this time. I would advise you to open an issue with the upstream vulnerability scanner to either ignore this dependency specifically, or better yet, ignore all dependencies which have a scope of `provided` from consideration.

**Reply by expertesantos** [reply]
Ok, thanks for your help
